hybrasyl/Objects/Creature.cs

```diff
@@ -854,6 +854,16 @@ public virtual bool UseCastable(Castable castObject, Creature target = null)
                 }
                 if (castObject.Effects?.Damage != null)
                 {
+                    if(this is User)
+                    {
+                        if (tar.MonsterDeathPileAllowedLooters.Count > 0 && !(tar.MonsterDeathPileAllowedLooters.Contains(Name))) continue;
+                        else if (tar.MonsterDeathPileAllowedLooters.Count == 0)
+                        {                            
+                            var theUser = this as User;
+                            if (theUser.Grouped) tar.MonsterDeathPileAllowedLooters = theUser.Group.Members.Select(user => user.Name).ToList();
+                            else tar.MonsterDeathPileAllowedLooters.Add(theUser.Name);
+                        }
+                    }
```

Comment:
all of this should be moved to `OnDeath()`

Reply:
I thought this might be applicable here to also handle attacking users, essentially the first attack "tagging"

Reply:
we have `LastHitter`; it might make sense for there to be `FirstHitter` to handle tagging. We may want to set that in `Damage()`. `OnDeath()` should still handle the allowed looters and should use `First/LastHitter` as needed.